File: frontend/src/types/session.ts
export interface Session {
  id: string;
  title: string;
  mode_id: string;
  status: "active" | "paused" | "completed";
  current_node_id: string | null;
  context_summary: string | null;
  created_at: string;
  updated_at: string;
}

export interface Mode {
  id: string;
  name: string;
  description: string;
  icon: string;
  question_categories: string[];
  initial_question: string;
}

File: backend/schemas/session.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class SessionCreate(BaseModel):
    mode_id: str
    title: Optional[str] = None


class SessionUpdate(BaseModel):
    title: Optional[str] = None
    status: Optional[str] = None


class SessionResponse(BaseModel):
    id: str
    title: str
    mode_id: str
    status: str
    current_node_id: Optional[str] = None
    context_summary: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SessionListResponse(BaseModel):
    sessions: list[SessionResponse]
